apps/site/.storybook/preview.tsx

```diff
@@ -3,9 +3,13 @@ import { withThemeByDataAttribute } from '@storybook/addon-themes';
 import type { Preview, ReactRenderer } from '@storybook/react';
 import { NextIntlClientProvider } from 'next-intl';
 
-import { STORYBOOK_MODES, STORYBOOK_SIZES } from '@/.storybook/constants';
 import { NotificationProvider } from '@/providers/notificationProvider';
 
+import {
+  STORYBOOK_MODES,
+  STORYBOOK_SIZES,
+} from '../../../.storybook/constants';
```

Comment:
FYI Storybook should exist only within the UI components package :)

Reply:
Storybook is in use in both the UI components, and the `app/site`. To prevent duplicate code, I moved it to the repo root (like Tailwind), what do you have in mind?

Reply:
Which components are on app/site that use Storybook? IMO, Storybook should **ONLY** exist for the UI components package.

Because semantically, that's what it is supposed to be used for. For describing components. I think we use it on apps/site only for visual regression, for whose I don't think they should be storybooks but just snapshots. We should probably make that change before we make this package then.

Reply:
We can still use Chromatic for that: https://www.chromatic.com/features/visual-test.

IMO you can do that on this PR, decoupling stories from website-only components (which actually are combination of the base components) and simply use Cypress or Node.js or whatever for the visual regression tests with Chromatic.

Reply:
I get it, I just believe this gives an excellent opportunity for so; This PR is already going to be huge, and I would say that ensuring that Storybooks only exist for actual UI components (instead of composite components/fragments/sections of a page) and that we still ensure visual regression on everything is definitely part of the goal of this package IMO.

We don't need to rush here, so if the issue is time, take your time :)

Reply:
> The current Storybook configuration in /apps/site differs from the one in packages/ui-components because the ui-components package doesn't rely on Next.js internally.

That doesn't make sense. The current Storybook configuration is Next.js-agnostic.

Reply:
> If we were to move the Storybook setup into ui-components now, we'd have to introduce Next.js dependencies, as the configuration in apps/site depends on Next.js internals. The goal of this package is to not have Next.js dependencies.

Could you elaborate where you saw that? I'm pretty sure it is not.

Reply:
> By "Next.js-free," I mean the removal of dependencies like next-intl, Link, and similar Next.js-specific elements.

You do know these lines are to "mock" components that use such dependencies, and that the ui-components library does not use these said dependencies, since the Link component for example, will now use Dependency Injection... Also, the Storybook should only exist for UI components, meaning there is nothing Next.js specific; I'm not sure where the confusion is here.

Sorry, but I'm adamant that Storybook should only be for ui-components, I won't approve this PR without that :/

Reply:
> A more effective approach IMO would be to gradually migrate components to the packages/ui-components package while removing Next.js-related elements from their Storybook configurations. Once all configurations are free of Next.js and consolidated in packages/ui-components, transferring the entire Storybook setup to this package will be much smoother.

I don't think we need a gradative migration on the main branch; as mentioned before, nothing stops you from making a feature branch and gradatively migrating the components; there's no rush to getting this done, and I doubt that conflicts will arise over time...

I'll defer to the rest of the @nodejs/nodejs-website if they are fine with merging this, as it will complicate things further with two different Storybooks.

I only want that we have a dedicated Storybook for UI components and that's it; No other Storybook, not mess needed. I won't block this PR, but I'm expressing my discontent.